test/bin/stress_testing.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash
+
+set -uo pipefail
+
+SCRIPT_NAME="stress_testing.sh"
+VERSION="0.2.0"
+
+INTERFACE=$(ip route | awk '/default/{print $5}')
+
+# script info
+function version {
+	echo -e "${VERSION}"
+}
+
+function help {
+  echo -e "USAGE: "
+  echo -e "       ./${SCRIPT_NAME} ACTION SCENARIO [VALUE]"
+  echo -e "       ./${SCRIPT_NAME} -h | --help"
+  echo -e "       ./${SCRIPT_NAME} -V | --version\n"
+  echo -e "ACTION:"
+  echo -e "  -e, --enable SCENARIO VALUE     Enable a scenario"
+  echo -e "  -d, --disable SCENARIO          Disable a scenario\n"
+  echo -e "  -g, --get SCENARIO              Returns the current real value for the scenario\n"
+  echo -e "SCENARIO:"
+  echo -e "  -s, --scenario { cpu | memory | disk | bandwith | packet_loss | latency } [ VALUE ]"
+  echo -e "        cpu: limit the usage cpu % to the value"
+  echo -e "        memory: limit the available memory to the value"
+  echo -e "        disk: limit the free space to the value"
+  echo -e "        bandwidth: limit the max download and upload bandwith to the value"
+  echo -e "        packet_loss: set packet loss % to the value"
+  echo -e "        latency: limit minimun latency to the value\n"
+  echo -e "VALUE:"
+  echo -e "  -v, --value VALUE     The metric value to force\n"
+  echo -e "EXAMPLE: "
+  echo -e "        ./${SCRIPT_NAME} --enable --scenario latency --value 100ms"
+}
+
+# check params
+function arg_precheck {
+  # no params check
+  if [ $# -eq 0 ]; then
+    help
+    exit 1
+  fi
+
+  # 1 param check
+  if [ $# -eq 1 ]; then
+    if [ "$*" == "-h" ] || [ "$*" == "--help" ]; then
+      help
+      exit 0
+    fi
+	fi
+
+  # scenario param check
+  if [ $# -eq 2 ]; then
+    echo -e "ERROR: scenario param is missing"
+    help
+    exit 1
+  fi 
+
+  # value param check
+  if [ $# -eq 4 ]; then
+    echo -e "ERROR: value param is missing"
+    help
+    exit 1
+  fi 
+}
+
+function arg_postcheck {
+  if [ "${ACTION}" == "enable" ]; then 
+    if [ "${TOTAL_PARAMS}" -eq 1 ]; then
+      echo -e "ERROR: scenario and value params are missing"
+      help
+      exit 1
+    fi
+  elif [ "${ACTION}" == "disable" ]; then
+    if [ "${TOTAL_PARAMS}" -lt 3 ]; then
+      echo -e "ERROR: scenario param is missing"
+      help
+      exit 1
+    elif [ "${TOTAL_PARAMS}" -gt 3 ]; then
+      echo -e "ERROR: value param must be removed"
+      help
+      exit 1
+    fi
+  fi
+}
+
+function tbd {
+  echo -e "This action is not implemented yet"
+}
+
+# bandwidth scenario
+function bandwidth_scenario {
+  bandwidth_scenario_ready
+  if [ "${ACTION}" == "enable" ]; then
+    enable_bandwidth
+  elif [ "${ACTION}" == "disable" ]; then
+    disable_bandwidth
+  elif [ "${ACTION}" == "get" ]; then
+    get_bandwidth
+  fi
+}
+
```

Comment:
dnf can not be used to install packages on a RHEL for Edge os-tree based system.
These tools need to exist on the MicroShift host, and would therefore need to be added from a [[packages]] stanza in the toml blueprint files. This script could check that the dependencies are available but can't assume it's able to install them.

The tests/image-blueprints directory has the toml files used to build images (See the "Image Blueprints" section of the README).